### DISCUSSION TITLE
Welcome to lightweight-registration Discussions!

### DISCUSSION THREAD

**Discussion by greglandrum** [rinikerlab/lightweight-registration#63] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jw-feng** [comment]
Thanks for making lwreg available. I am evaluating it as a solution to register a few hundred small molecules to generate unique IDs. Duplicates result in RegistrationFailureReasons.DUPLICATE. Any thoughts about incrementing a batch number and returning a duplicated molecule message?

**Reply by greglandrum** [reply]
@jw-feng Handling batches (or samples) is clearly important functionality for a registration system intended for handling things made in the real world, but I don't think it fits into lwreg itself.

However, it would be quite straightforward to add a thin layer yourself to do this.

If you pass the optional argument 'fail_on_duplicate=False` to the `register()` function, it will return the molregno of the already registered structure instead of raising an error:

```
In [5]: lwreg.register(smiles='CCC')
Out[5]: 1

In [6]: lwreg.register(smiles='CCO')
Out[6]: 2

In [7]: lwreg.register(smiles='CCOC')
Out[7]: 3

In [8]: lwreg.register(smiles='CCC',fail_on_duplicate=False)
Out[8]: 1
```

So you could have a separate table which maps a molregno to a batch number and just add a new entry every time you register with the appropriate batch id.
